terraspace/app/stacks/shared/auth.tf

```diff
@@ -122,3 +122,90 @@ EOF
     create_before_destroy = true
   }
 }
+
+
+resource "aws_iam_policy" "dynamodb_v1_blocks_policy" {
+  name        = "dynamodb-${var.v1_blocks_table.name}-policy"
+  description = "Policy for allowing all Dynamodb Actions for blocks table"
+  policy      = <<EOF
+{  
+  "Version": "2012-10-17",
+  "Statement":[{
+    "Effect": "Allow",
+    "Action": [
+     "dynamodb:BatchGetItem",
+     "dynamodb:GetItem",
+     "dynamodb:Query",
+     "dynamodb:Scan",
+     "dynamodb:BatchWriteItem",
+     "dynamodb:PutItem",
+     "dynamodb:UpdateItem",
+     "dynamodb:DeleteItem"
+    ],
+    "Resource": "${aws_dynamodb_table.v1_blocks_table.arn}"
+   }
+  ]
+}
+EOF
+
+  lifecycle {
+    create_before_destroy = true
```

Comment:
In order for `create_before_destroy` to operate as expected, this `resource`s name needs to be unique.  I don't see that happening here.

Reply:
Useful for migrating from one policy to a new one (with different name) without removing existing policies from roles